core/src/main/java/hudson/model/DirectoryBrowserSupport.java

```diff
@@ -356,33 +353,33 @@ private static String createBackRef(int times) {
-        ZipOutputStream zos = new ZipOutputStream(outputStream);
-        zos.setEncoding(System.getProperty("file.encoding")); // TODO JENKINS-20663 make this overridable via query parameter
-        for (String n : dir.list(glob.length() == 0 ? "**" : glob)) {
-            String relativePath;
-            if (glob.length() == 0) {
-                // JENKINS-19947: traditional behavior is to prepend the directory name
-                relativePath = dir.getName() + '/' + n;
-            } else {
-                relativePath = n;
-            }
-            // In ZIP archives "All slashes MUST be forward slashes" (http://pkware.com/documents/casestudies/APPNOTE.TXT)
-            // TODO On Linux file names can contain backslashes which should not treated as file separators.
-            //      Unfortunately, only the file separator char of the master is known (File.separatorChar)
-            //      but not the file separator char of the (maybe remote) "dir".
-            ZipEntry e = new ZipEntry(relativePath.replace('\\', '/'));
-            VirtualFile f = dir.child(n);
-            e.setTime(f.lastModified());
-            zos.putNextEntry(e);
-            InputStream in = f.open();
-            try {
-                Util.copyStream(in, zos);
-            } finally {
-                IOUtils.closeQuietly(in);
+        try (ZipOutputStream zos = new ZipOutputStream(outputStream)) {
+            zos.setEncoding(System.getProperty("file.encoding")); // TODO JENKINS-20663 make this overridable via query parameter
+            for (String n : dir.list(glob.length() == 0 ? "**" : glob)) {
+                String relativePath;
+                if (glob.length() == 0) {
+                    // JENKINS-19947: traditional behavior is to prepend the directory name
+                    relativePath = dir.getName() + '/' + n;
+                } else {
+                    relativePath = n;
+                }
+                // In ZIP archives "All slashes MUST be forward slashes" (http://pkware.com/documents/casestudies/APPNOTE.TXT)
+                // TODO On Linux file names can contain backslashes which should not treated as file separators.
+                //      Unfortunately, only the file separator char of the master is known (File.separatorChar)
+                //      but not the file separator char of the (maybe remote) "dir".
+                ZipEntry e = new ZipEntry(relativePath.replace('\\', '/'));
+                VirtualFile f = dir.child(n);
+                e.setTime(f.lastModified());
+                zos.putNextEntry(e);
+                InputStream in = f.open();
+                try {
+                    Util.copyStream(in, zos);
+                } finally {
+                    IOUtils.closeQuietly(in);
+                }
+                zos.closeEntry();
             }
-            zos.closeEntry();
         }
-        zos.close();
```

Comment:
And again. Unlikely there is a real reason for the original code without issue handling